tensorflow_io/core/kernels/azfs_kernels.cc

```diff
@@ -608,109 +692,189 @@ Status AzBlobFileSystem::DeleteDir(const std::string &dirname) {
+static void RenameFile(const TF_Filesystem* filesystem, const char* src,
+                       const char* dst, TF_Status* status) {
   std::string src_account, src_container, src_object;
-  TF_RETURN_IF_ERROR(
-      ParseAzBlobPath(src, false, &src_account, &src_container, &src_object));
-  std::string target_account, target_container, target_object;
-  TF_RETURN_IF_ERROR(ParseAzBlobPath(target, false, &target_account,
-                                     &target_container, &target_object));
+  ParseAzBlobPath(src, false, &src_account, &src_container, &src_object,
+                  status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
+  }
+
+  std::string dst_account, dst_container, dst_object;
+  ParseAzBlobPath(dst, false, &dst_account, &dst_container, &dst_object,
+                  status);
+  if (TF_GetCode(status) != TF_OK) {
+    return;
+  }
 
-  if (src_account != target_account) {
-    return errors::Unimplemented(
-        "Couldn't rename ", src, " to ", target,
-        ": moving files between accounts is not supported.");
+  if (src_account != dst_account) {
+    std::string error_message =
+        absl::StrCat("Couldn't rename ", src, " to ", dst,
+                     ": moving files between accounts is not supported");
+    TF_SetStatus(status, TF_UNIMPLEMENTED, error_message.c_str());
+    return;
   }
 
   auto blob_client = CreateAzBlobClientWrapper(src_account);
 
-  blob_client.start_copy(src_container, src_object, target_container,
-                         target_object);
+  blob_client.start_copy(src_container, src_object, dst_container, dst_object);
   if (errno != 0) {
-    return errors::Internal("Failed to start rename from ", src, " to ", target,
-                            " (", errno_to_string(), ")");
+    std::string error_message =
+        absl::StrCat("Failed to start rename from ", src, " to ", dst, " (",
+                     errno_to_string(), ")");
+    TF_SetStatus(status, TF_INTERNAL, error_message.c_str());
+    return;
   }
 
   // Wait until copy completes
   // Status can be success, pending, aborted or failed
   std::string copy_status;
   do {
-    const auto target_blob_property =
-        blob_client.get_blob_property(target_container, target_object);
-    copy_status = target_blob_property.copy_status;
+    const auto dst_blob_property =
+        blob_client.get_blob_property(dst_container, dst_object);
+    copy_status = dst_blob_property.copy_status;
   } while (copy_status.find("pending") == 0 && !copy_status.empty());
 
   if (copy_status.find("success") == std::string::npos) {
-    return errors::Internal("Process of renaming resulted in status of ",
-                            copy_status, " when renaming ", src, " to ",
-                            target);
+    std::string error_message =
+        absl::StrCat("Process of renaming resulted in status of ", copy_status,
+                     " when renaming ", src, " to ", dst);
```

Comment:
absl::StrCat("Process of renaming from , src, " to ", dst, " resulted in status of ", copy_status);

Reply:
Thanks @kvignesh1420 . Updated.